src/xo-app/settings/remotes/index.js

```diff
@@ -39,6 +42,31 @@ class AbstractRemote extends Component {
     return editRemote(remote, {name})
   }
 
+  _test = () => {
+    const { remote } = this.props
+    testRemote(remote).then(answer => {
+      const title = <span>
+        <Icon icon={answer.success ? 'success' : 'error'} />
+        {' '}
+        {answer.success ? _('remoteTestSuccess') : _('remoteTestFailure')} {remote.name}
```

Comment:
I'm not sure to catch you, I searched the react-intl doc for a kind of "switch" pattern in the translation message expressions but found nothing so far...